src/components/Row/Row.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import Square from '../Square/Square';
+import Wall from '../Wall/Wall';
+
+function Row(props) {
```

Comment:
import React from 'react';
import Square from '../Square/Square';
import Wall from '../Wall/Wall';

const Row = (props) => {
  const {
    boardSize,
    rowIndex,
    onPlayerClick,
    highlightedSquares,
    movePlayer,
    players,
    onWallHover,
    hoveredWalls,
  } = props;

  const renderSquare = (colIndex) => {
    const player = players.find(
      (player) => player.position.row === rowIndex && player.position.col === colIndex
    )?.name || '';
    
    const isHighlighted = highlightedSquares.some(
      (square) => square.row === rowIndex && square.col === colIndex
    );

    return (
      <Square
        key={`square-${rowIndex}-${colIndex}`}
        id={`square-${rowIndex}-${colIndex}`}
        rowIndex={rowIndex}
        colIndex={colIndex}
        player={player}
        isHighlighted={isHighlighted}
        onPlayerClick={onPlayerClick}
        highlightedSquares={highlightedSquares}
        movePlayer={movePlayer}
      />
    );
  };

  const renderWall = (colIndex) => {
    const vWallId = `vwall-${rowIndex}-${colIndex}`;
    return (
      <Wall
        key={vWallId}
        id={vWallId}
        orientation="vertical"
        onWallHover={onWallHover}
        isHovered={hoveredWalls.includes(vWallId)}
      />
    );
  };

  const renderRowContent = () => {
    const rowContent = [];
    for (let colIndex = 0; colIndex < boardSize; colIndex++) {
      rowContent.push(renderSquare(colIndex));
      if (colIndex !== boardSize - 1) {
        rowContent.push(renderWall(colIndex));
      }
    }
    return rowContent;
  };

  return (
    <tr key={`row-${rowIndex}`} id={`row-${rowIndex}`}>
      {renderRowContent()}
    </tr>
  );
};

export default Row;

Reply:
You can use the above code. this more SoC-clean code principle code pieces